modules/browsiRtdProvider.js

```diff
@@ -155,36 +167,88 @@ function getKVObject(p, keyName) {
 /**
  * check if placement id matches one of given ad units
  * @param {number} id placement id
- * @param {Object[]} allSlots google slots on page
+ * @param {Object} slot google slot
  * @param {string[]} whitelist ad units
  * @return {boolean}
  */
-export function isIdMatchingAdUnit(id, allSlots, whitelist) {
+export function isIdMatchingAdUnit(id, slot, whitelist) {
   if (!whitelist || !whitelist.length) {
     return true;
   }
-  const slot = allSlots.filter(s => s.getSlotElementId() === id);
-  const slotAdUnits = slot.map(s => s.getAdUnitPath());
-  return slotAdUnits.some(a => whitelist.indexOf(a) !== -1);
+  const slotAdUnits = slot.getAdUnitPath();
+  return whitelist.indexOf(slotAdUnits) !== -1;
 }
 
+/**
+ * get GPT slot by placement id
+ * @param {string} id placement id
+ * @return {?Object}
+ */
+function getSlotById(id) {
+  const slots = getAllSlots();
+  if (!slots || !slots.length) {
+    return null;
+  }
+  return slots.filter(s => s.getSlotElementId() === id)[0] || null;
+}
+
+/**
+ * generate id according to macro script
+ * @param {string} macro replacement macro
+ * @param {string} id placement id
+ * @param {Object} slot google slot
+ * @return {?Object}
+ */
+function getMacroId(macro, id, slot) {
+  if (macro) {
+    try {
+      const macroResult = evaluate(macro, slot.getSlotElementId(), slot.getAdUnitPath(), (match, p1) => {
+        return (p1 && slot.getTargeting(p1).join('_')) || 'NA';
+      });
+      console.log(`${slot.getSlotElementId()} macro result: ${macro} => ${macroResult}`);
```

Comment:
If this is a log message you want to keep to help with debugging purposes, can you please use the `utils.logMessage` or `utils.logInfo` methods instead of just `console.log`?  This would help to flag it properly in the browser console logs as something related to Prebid.

If you don't want to keep it long-term, can you please remove it?

Reply:
log removed